empress/support_files/js/color-options-handler.js

```diff
@@ -0,0 +1,273 @@
+define(["underscore", "Colorer", "util"], function (_, Colorer, util) {
+    var TotalColorOptionsHandlers = 0;
+
+    function ColorOptionsHandler(container, enableContinuousColoring = false) {
+        // add count
+        TotalColorOptionsHandlers += 1;
+
+        this.container = container;
+        this.observers = [];
+        this.defaultColor = "discrete-coloring-qiime";
+        this.defaultReverseChk = false;
+        // create unique num
+        this.uniqueNum = TotalColorOptionsHandlers;
+        this.enableContinuousColoring = enableContinuousColoring;
+
+        // Add a row for choosing the color map
+        var colormapP = this.container.appendChild(document.createElement("p"));
+        var colormapLbl = colormapP.appendChild(
+            document.createElement("label")
+        );
+        colormapLbl.innerText = "Color Map";
+        var colormapSC = colormapP.appendChild(document.createElement("label"));
+        colormapSC.classList.add("select-container");
+        this.colormapSelector = document.createElement("select");
+        Colorer.addColorsToSelect(this.colormapSelector);
+        colormapSC.appendChild(this.colormapSelector);
+        this.colormapSelector.id =
+            "color-options-handler-" + this.uniqueNum + "-colormap-select";
+        colormapLbl.setAttribute("for", this.colormapSelector.id);
+
+        // Add a row for choosing whether the color scale should
+        // be reversed
+        var reverseColormapP = this.container.appendChild(
+            document.createElement("p")
+        );
+        var reverseColormapLbl = reverseColormapP.appendChild(
+            document.createElement("label")
+        );
+        reverseColormapLbl.innerText = "Reverse Color Map";
+        this.reverseColormapCheckbox = reverseColormapP.appendChild(
+            document.createElement("input")
+        );
+        this.reverseColormapCheckbox.id =
+            "color-options-hanlder-" + this.uniqueNum + "-reverse-chk";
```

Comment:
```suggestion
            "color-options-handler-" + this.uniqueNum + "-reverse-chk";
```

Looks like there are still 5 occurrences of `hanlder` throughout this file; could you please fix these? I'm sorry to be picky, but it's going to make the code harder to work with if we can't rely on IDs to be properly named.